ledger/narwhal/batch-certificate/src/lib.rs

```diff
@@ -82,6 +72,20 @@ impl<N: Network> BatchCertificate<N> {
         // Return the batch certificate.
         Ok(Self { batch_header, signatures })
     }
+
+    pub fn check_signature_basic(batch_header: &BatchHeader<N>, signatures: &IndexSet<Signature<N>>) -> Result<()> {
```

Comment:
Could you document this function? It should be clear what it does compared to the checks in `BatchCertificate::from`. 

Maybe you could document `BatchCertificate::from` a little better as well.